sponsorship_compassion/models/contracts_report.py

```diff
@@ -0,0 +1,226 @@
+                 ('gender', '=', 'F')])
+            if partner.is_church:
+                for member in partner.member_ids:
+                    total += self.env['compassion.child'].search_count(
+                        [('partner_id', '=', member.id),
+                         ('gender', '=', 'F')])
+            partner.sr_nb_girl = total
+
+    @api.multi
+    def _compute_time_scp(self):
+        def get_time_in_scp(_partner):
+            total = 0
+            for contract in _partner.sponsorship_ids.filtered(
+                    lambda s: s.is_active is True):
+                today = date.today()
+                contract_start_date = fields.Date.from_string(
+                    contract.start_date)
+                delta = today - contract_start_date
+
+                nb_weeks = delta.days / 7
+                country = contract.child_id.field_office_id
+                total += nb_weeks * country.icp_hours_week
+            return total
+
+        for partner in self:
+            time_to_scp = get_time_in_scp(partner)
+
+            if partner.is_church:
+                for member in partner.member_ids:
+                    time_to_scp += get_time_in_scp(member)
+            partner.sr_time_scp = time_to_scp
+
+    @api.multi
+    def _compute_meal(self):
+        def get_nb_meal(_contract):
+            today = date.today()
+            contract_start_date = fields.Date.from_string(
+                _contract.start_date)
+            delta = today - contract_start_date
+
+            nb_weeks = delta.days / 7
+            country = contract.child_id.field_office_id
+            return nb_weeks * country.icp_meal_week
+
+        for partner in self:
+            sr_meal = 0
+            for contract in partner.sponsorship_ids:
+                sr_meal += get_nb_meal(contract)
+
+            if partner.is_church:
+                for member in partner.member_ids:
+                    for contract in member.sponsorship_ids:
+                        sr_meal += get_nb_meal(contract)
+            partner.sr_meal = sr_meal
+
+    @api.multi
+    def _compute_sr_payment(self):
+        def get_sum_invoice(_partner):
+            invoices = self.env['account.invoice.line'].search([
+                ('partner_id', '=', _partner.id),
+                ('last_payment', '<', _partner.end_period),
+                ('last_payment', '>', _partner.start_period)
+            ])
+            return sum(invoices.mapped('price_unit'))
+
+        for partner in self:
+            sr_payment = get_sum_invoice(partner)
+            if partner.is_church:
+                for member in partner.member_ids:
+                    sr_payment += get_sum_invoice(member)
+            partner.sr_payment = sr_payment
+
+    @api.multi
+    def _compute_sr_gift(self):
+        def get_nb_gift(_partner):
+            return self.env['account.invoice'].search_count(
+                [('partner_id', '=', _partner.id),
+                 ('invoice_type', '=', 'gift'),
```

Comment:
Add filter ('type', '=', 'out_invoice') to avoid taking supplier invoices.
Add filter ('state', '=', 'paid') otherwise you will include open invoices.

Furthermore, field `invoice_type` is not present in the module! This will fail on the tests or databases without sponsorship_switzerland field. You should therefore move the field from sponsorship_switzerland to sponsorship_compassion if you need it. Don't forget to add migration script otherwise it will break the data. You can see an example here: https://github.com/CompassionCH/compassion-switzerland/blob/devel/partner_compassion/migrations/10.0.1.2.0/pre-migration.py#L19